Sources/Regex/Atom.swift

```diff
@@ -404,19 +404,15 @@ extension Atom: _ASTPrintable {
 
 extension Atom.EscapedBuiltin {
   var characterClass: CharacterClass? {
-    // TODO: Hamish, can you take over?
-
     switch self {
-    case .singleDataUnit: fatalError("TODO")
-
     case .decimalDigit:    return .digit
     case .notDecimalDigit: return .digit.inverted
 
     case .horizontalWhitespace: return .horizontalWhitespace
     case .notHorizontalWhitespace:
       return .horizontalWhitespace.inverted
 
-    case .notNewline: fatalError("TODO")
+    case .notNewline: return .newlineSequence.inverted
```

Comment:
Just to double check, there's `\R`, `\n`, and `\N`. Do we handle all 3? Is `\N` the same as `\R` inverted?

Reply:
Yes we handle all 3, and I believe `\N` is indeed `\R` inverted, or at least it is if we're recognizing all the unicode line endings as characters not matched by `.` when not in multi-line mode

Reply:
But that's contingent on semantic matching mode as well as options, right? Is `\N` the inverse of `\R` or `\n`?

Reply:
Hmm yeah it seems the newlines we recognize would be affected by the `(*CR)`, `(*LF)`, `(*CRLF)`, `(*ANYCRLF)`, `(*ANY)` and `(*NUL)` sequences. PCRE also appears to default to recognizing only `\n`, at least if a convention isn't provided to `pcre2_compile`. Really we'd want to track the newline convention we're matching against for the `any` character class, though perhaps for now we should just leave it as the inverse of `\n`?

Reply:
We will have options that govern what is in a character class, at a semantic level, but that shouldn't affect parsing or really code in the parser (outside of us parsing the options and/or faithfully ferrying them along). IIRC, `\N` is normally the inverse of `\n`, but we can treat inverses as distinct classes syntactically.